go-controller/pkg/informer/informer.go

```diff
@@ -0,0 +1,176 @@
+// Package informer provides a wrapper around a client-go SharedIndexInformer
+// for event handling. It removes a lot of boilerplate code that would be required
+// to use workqueues.
+package informer
+
+import (
+	"fmt"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+)
+
+// EventHandler is an event handler that responds to
+// Add/Delete events from the informer cache
+type EventHandler interface {
+	Run(threadiness int, stopChan <-chan struct{}) error
+	GetIndexer() cache.Indexer
+}
+
+type eventHandler struct {
+	name           string
+	informer       cache.SharedIndexInformer
+	deletedIndexer cache.Indexer
+	workqueue      workqueue.RateLimitingInterface
+	add            func(obj interface{}) error
+	delete         func(obj interface{}) error
+}
+
+// NewDefaultEventHandler returns a default event handler
+func NewDefaultEventHandler(name string, informer cache.SharedIndexInformer, addFunc, deleteFunc func(obj interface{}) error) EventHandler {
+	e := &eventHandler{
+		name:           name,
+		informer:       informer,
+		deletedIndexer: cache.NewIndexer(cache.DeletionHandlingMetaNamespaceKeyFunc, cache.Indexers{}),
+		workqueue:      workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter()),
+		add:            addFunc,
+		delete:         deleteFunc,
+	}
+	informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			e.enqueue(obj)
```

Comment:
I think you should remove `obj` from the deleted set here. (and possible in Update as well...)

Reply:
Good call. I think it's better to remove it after the event is processed in the `syncHandler` function so I'll go ahead and add that.